reference/test/matrix/ell_kernels.cpp

```diff
@@ -0,0 +1,346 @@
+   and/or other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its contributors
+   may be used to endorse or promote products derived from this software
+   without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+#include <core/matrix/ell.hpp>
+
+
+#include <gtest/gtest.h>
+
+
+#include <core/base/exception.hpp>
+#include <core/base/exception_helpers.hpp>
+#include <core/base/executor.hpp>
+#include <core/matrix/dense.hpp>
+#include <core/test/utils/assertions.hpp>
+
+
+namespace {
+
+
+class Ell : public ::testing::Test {
+protected:
+    using Mtx = gko::matrix::Ell<>;
+    using Vec = gko::matrix::Dense<>;
+
+    Ell()
+        : exec(gko::ReferenceExecutor::create()),
+          mtx1(Mtx::create(exec, 2, 3, 3, 2)),
+          mtx2(Mtx::create(exec, 2, 3, 3, 16))
+    {
+        Mtx::value_type *v = mtx1->get_values();
+        Mtx::index_type *c = mtx1->get_col_idxs();
+        Mtx::index_type n = mtx1->get_max_nonzeros_per_row();
+        c[0] = 0;
+        c[1] = 1;
+        c[2] = 1;
+        c[3] = 0;
+        c[4] = 2;
+        c[5] = 0;
+        v[0] = 1.0;
+        v[1] = 5.0;
+        v[2] = 3.0;
+        v[3] = 0.0;
+        v[4] = 2.0;
+        v[5] = 0.0;
+
+        v = mtx2->get_values();
+        c = mtx2->get_col_idxs();
+        n = mtx2->get_max_nonzeros_per_row();
+        gko::size_type padding = mtx2->get_padding();
+        for (gko::size_type i = 0; i < n * padding; i++) {
+            c[i] = 0;
+            v[i] = 0.0;
+        }
+        c[0] = 0;
+        c[1] = 1;
+        c[0 + 1 * padding] = 1;
+        c[0 + 2 * padding] = 2;
+        v[0] = 1.0;
+        v[1] = 5.0;
+        v[0 + 1 * padding] = 3.0;
+        v[0 + 2 * padding] = 2.0;
+    }
+
+    std::shared_ptr<const gko::Executor> exec;
+    std::unique_ptr<Mtx> mtx1;
+    std::unique_ptr<Mtx> mtx2;
+};
+
+
+TEST_F(Ell, AppliesToDenseVector)
+{
+    auto x = gko::initialize<Vec>({2.0, 1.0, 4.0}, exec);
+    auto y = Vec::create(exec, 2, 1, 1);
+
+    mtx1->apply(x.get(), y.get());
+
+    EXPECT_EQ(y->at(0), 13.0);
+    EXPECT_EQ(y->at(1), 5.0);
```

Comment:
You can use:

```c++
ASSERT_MTX_NEAR(y, l({13.0, 5.0}), 0.0);
```

instead of multiple expects on matrix elements.

Reply:
Fix it.